### DISCUSSION TITLE
Add search capabilities to ghostty.org

### DISCUSSION THREAD

**Discussion by shortcuts** [ghostty-org/ghostty#3195] [Feature Requests, Ideas]
_(Initially proposed in https://github.com/ghostty-org/website/pull/137, [see POC](https://website-git-fork-shortcuts-main-ghostty.vercel.app/docs)), kind of similar discussion: https://github.com/ghostty-org/ghostty/discussions/3022_

----

I was browsing the docs for `shell integrations with fish` and realized there was no search capabilities, and was wondering if the team already had solutions in mind for future implementations.

While the project has only been public for a few hours, and might not be indexed by global search engines yet, having a searchable documentation from the website provides a great user experience.

## Proposal

Obviously I'm biased here because I work at Algolia and I'm one of the main contributor of [DocSearch](https://docsearch.algolia.com/) ([repository](https://github.com/algolia/docsearch)) but I like to help setting up search instances for tools that I use, so here's a small POC for Ghostty:

| Desktop | Mobile |
|--------|--------|
| <img width="1490" alt="Screenshot 2024-12-26 at 22 59 35" src="https://github.com/user-attachments/assets/f11bb0e1-cbc8-44e8-a9fe-12c220058b30" /> | <img width="383" alt="Screenshot 2024-12-26 at 23 00 22" src="https://github.com/user-attachments/assets/cad9af1a-3251-4ff6-a640-b6efefe64081" /> |
| <img width="1484" alt="Screenshot 2024-12-26 at 22 59 55" src="https://github.com/user-attachments/assets/a6f765af-03af-4d05-9bce-c4d59f30eb7d" /> | <img width="375" alt="Screenshot 2024-12-26 at 23 01 01" src="https://github.com/user-attachments/assets/d14d2299-7b44-4e79-84fb-4ba6483fd534" /> |

You can also browse the live preview here: https://website-git-fork-shortcuts-main-ghostty.vercel.app/docs

Disclaimer: The API key is search-only and public, no worries about disclosing it in the POC. I also did not managed to match the website fonts

### Infrastructure costs

DocSearch is free for open-source projects, no additional costs on your side.

### Maintenance costs

I can help finishing the POC and improving the indexing logic/search relevancy based on your expectations.

**Comment by Kerollmops** [comment]
Hey 👋

I recommend [using Meilisearch](https://meilisearch.com). It's also open source 😉 You can probably follow [what Tauri is doing](https://github.com/tauri-apps/meilisearch-docsearch).

Note, I am the co-founder 😉

**Reply by annesnour03** [reply]
> Hey 👋
> 
> I recommend [using Meilisearch](https://meilisearch.com). It's also open source 😉 You can probably follow [what Tauri is doing](https://github.com/tauri-apps/meilisearch-docsearch).

I think it's appropriate if you note that you're one of the co-founders 😃

**Comment by BrandonRomano** [comment]
I am generally in favor of an Algolia based approach; largely is as I am familiar with it and know it will solve for our use case.  The fact that there are no costs to OSS projects seems pretty great to me.

My vote here is not authoritative; but I wanted to share where I am leaning. Others should still weigh in here if there are any opinions.

@shortcuts thanks for opening this discussion with the POC 👏

**Reply by goproslowyo** [reply]
It'd be neat to revisit the ghostty-org/website#137 PR that @shortcuts opened if Algolia is the one. It'd be nice to get something since ease of finding things like config options, other docs, release notes, etc is one of the things new people have been having a hard time with. 

It's a quick win and it'd help a great deal of newcomers out and will only get better as PRs to improve the docs on the website are merged.

It's also top 10 in the ideas/issue triage discussions :)

![image](https://github.com/user-attachments/assets/644a75b4-9c4d-42d1-b15d-066d6ad08673)

**Comment by aretrace** [comment]
How about something simpler like [Pagefind](https://pagefind.app/)

**Comment by sherifnada** [comment]
@BrandonRomano would love to help here! Happy to go with the proposed algolia approach, or something even lighter weight like Pagefind. Anything the community can do to help drive a consensus/decision on this?

**Comment by vladavoX** [comment]
Started using Ghostty today, and immediately ran into the missing search in the docs. I think Pagefind would be a simple and solid solution.

**Reply by jcollie** [reply]
What version did you install? Search was added in 1.3. https://ghostty.org/docs/install/release-notes/1-3-0#scrollback-search

**Reply by 00-kat** [reply]
This discussion is about ghostty.org, not the Ghostty app, lol. (Note “ghostty.org” in the title and “missing search <ins>in the docs</ins>”—emphasis mine—in vladavoX's comment.)

**Reply by vladavoX** [reply]
My comment is correct, this discussion is about `ghostty.org/docs` missing search feature in docs, and I've said that **missing search in the docs**, and I mean it, in docs, not the app

<img width="1631" height="328" alt="image" src="https://github.com/user-attachments/assets/2390e6b7-36eb-4b78-89da-b06f9f9d0bc0" />

**Reply by jcollie** [reply]
Hah, serves me right for answering these on mobile.
